olive/passes/onnx/conversion.py

```diff
@@ -195,13 +195,17 @@ def _export_pytorch_model(
 
             # handle dummy inputs for model with past, which has past_key_values
             # match input names in `past_key_values.(hidden_layer_num).(key|value)` pattern(llama2 case)
-            # or `past_key_values.(hidden_layer_num)` pattern (phi2 case)
+            # or `past_key_(hidden_layer_num)`| `past_value_(hidden_layer_num)` pattern (phi2 case)
+            import re
+
             for name, dm_input in dummy_inputs.items():
+                # the `past_key_values` is the argument name from huggingface model class
+                # which is independent of the kv-related variables in input list
                 if name == "past_key_values" and isinstance(dm_input, list):
-                    key_value_names = [f"{name}.{idx}" for idx in range(len(dm_input))]
+                    kv_name_regrex_strs = [rf"past_(.*)(.|_){idx}(.(key|value))?" for idx in range(len(dm_input))]
                     if all(
-                        any(key_value_name in input_name for input_name in input_names)
-                        for key_value_name in key_value_names
+                        any(re.match(kv_r_str, input_name) for input_name in input_names)
```

Comment:
regex building/match is pretty time consuming compared with simply string comparison.

Reply:
Updated. Also see comments:
```
        # In the case of dynamo exporting, user do not need to provide input names
        # for past_key_values, when exported to onnx, it will be expand to
        # `past_key_values.(hidden_layer_num).(key|value)` pattern(llama2 case)
        # or `past_(key|value)_(hidden_layer_num)` pattern (phi2 case)
        # after the expansion, user should provide the correct input names for inference
```